pkg/k8shandler/collection.go

```diff
@@ -241,6 +243,19 @@ func createOrUpdateCollectorServiceAccount(cluster *logging.ClusterLogging) erro
 		return fmt.Errorf("Failure creating Log collector privileged role binding: %v", err)
 	}
 
+	// create clusterrole for logcollector to retrieve metadata
+	clusterrules := utils.NewPolicyRules(
+		utils.NewPolicyRule(
+			[]string{""},
+			[]string{"pods", "namespaces"},
+			nil,
+			[]string{"get", "list", "watch"},
+		),
+	)
+	clusterRole, err := utils.CreateClusterRole("metadata-reader", clusterrules, cluster)
```

Comment:
Can you update this to follow the other role creation patterns for now?

Reply:
I'm not sure what you are asking here.  I added code to a reusable method that does the same thing as if it was inlined.